analytics/app/src/main/java/com/google/firebase/quickstart/analytics/java/MainActivity.java

```diff
@@ -104,28 +107,32 @@ protected void onCreate(Bundle savedInstanceState) {
             setUserFavoriteFood(userFavoriteFood);
         }
 
-
         // Create the adapter that will return a fragment for each image.
-        mImagePagerAdapter = new ImagePagerAdapter(getSupportFragmentManager(), IMAGE_INFOS);
+        mImagePagerAdapter = new ImagePagerAdapter(this, IMAGE_INFOS);
 
         // Set up the ViewPager with the pattern adapter.
-        mViewPager = binding.viewPager;
+        mViewPager =  (ViewPager2)findViewById(R.id.viewPager);
```

Comment:
We use [ViewBinding](https://developer.android.com/topic/libraries/view-binding) instead of `findViewById()` in this project.
```suggestion
        mViewPager =  binding.viewPager;
```